pynest/nest/lib/hl_api_helper.py

```diff
@@ -626,6 +627,26 @@ def model_deprecation_warning(model):
         warnings.warn('\n' + text)
 
 
+def serialize(data):
+    """
+    Serialize data for JSON.
```

Comment:
I don't see any connection to JSON here. The docstring should either explain what the function really does (i.e. convert all iterable types to simple lists) or the function should actually serialize to JSON (and then probably have a `_to_json` appended to its name). I don't see a problem with the latter (JSON is a Python standard library module anyway).